src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodServiceImpl.java

```diff
@@ -46,11 +48,14 @@ public BreedingMethodServiceImpl(final HibernateSessionProvider sessionProvider)
 	@Override
 	public BreedingMethodDTO getBreedingMethod(final Integer breedingMethodDbId) {
 		final Method methodEntity = this.daoFactory.getMethodDAO().getById(breedingMethodDbId);
-		return new BreedingMethodDTO(methodEntity);
+		if (!Objects.isNull(methodEntity)) {
+			return new BreedingMethodDTO(methodEntity);
+		}
+		return null;
```

Comment:
Returning null is a code smell. Better to make the method signature `Optional<BreedingMethodDTO> getBreedingMethod(final Integer breedingMethodDbId)`